src/com/ms/silverking/cloud/skfs/dir/serverside/FileDeletionWorker.java

```diff
@@ -8,47 +8,47 @@
 import com.ms.silverking.log.Log;
 
 class FileDeletionWorker implements Runnable {
-    private final BlockingQueue<File>    workQueue;
-    
-    private static final String    threadName = "FileDeletionWorker";
-    
-    public FileDeletionWorker() {
-        Thread    t;
-        
-        workQueue = new LightLinkedBlockingQueue<>();
-        t = new Thread(this, threadName);
-        t.setDaemon(true);
-        t.start();
-    }
+  private final BlockingQueue<File> workQueue;
+
+  private static final String threadName = "FileDeletionWorker";
+
+  public FileDeletionWorker() {
+    Thread t;
+
+    workQueue = new LightLinkedBlockingQueue<>();
+    t = new Thread(this, threadName);
+    t.setDaemon(true);
+    t.start();
+  }
 
-    @Override
-    public void run() {
-        while (true) {
-            try {
-                File    f;
-                
-                f = workQueue.take();
-                if (f != null) {
-                    if (f.exists() && !f.delete()) {
-                        Log.warningf("%s unable to delete %s", threadName, f.getAbsolutePath());
-                    }
-                }
-            } catch (Exception e) {
-                Log.logErrorWarning(e);
-            }
+  @Override
+  public void run() {
+    while (true) {
+      try {
+        File f;
+
+        f = workQueue.take();
+        if (f != null) {
+          if (f.exists() && !f.delete()) {
+            Log.warningf("%s unable to delete %s", threadName, f.getAbsolutePath());
+          }
         }
+      } catch (Exception e) {
+        Log.logErrorWarning(e);
+      }
     }
+  }
 
-    public void delete(File f) {
-        try {
-            workQueue.put(f);
-        } catch (InterruptedException e) {
-        }
+  public void delete(File f) {
+    try {
+      workQueue.put(f);
+    } catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)